test/e2e/set/scyllacluster/datainserter.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2021 ScyllaDB
+
+package scyllacluster
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/scylladb/scylla-operator/test/e2e/framework"
+
+	"github.com/gocql/gocql"
+	"github.com/scylladb/gocqlx/v2"
+	"github.com/scylladb/gocqlx/v2/table"
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const nRows = 10
+
+type DataInserter struct {
+	client            corev1client.CoreV1Interface
+	session           *gocqlx.Session
+	keyspace          string
+	table             *table.Table
+	data              []*TestData
+	replicationFactor int
+}
+
+type TestData struct {
+	Id   int    `db:"id"`
+	Data string `db:"data"`
+}
+
+func NewDataInserter(ctx context.Context, client corev1client.CoreV1Interface, sc *scyllav1.ScyllaCluster, replicationFactor int) (*DataInserter, error) {
+	if replicationFactor < 1 {
+		return nil, fmt.Errorf("replication factor can't be set to less than 1")
+	}
+
+	keyspace := strings.Replace(string(uuid.NewUUID()), "-", "", -1)
+	di := &DataInserter{
+		client:   client,
+		keyspace: keyspace,
+		table: table.New(table.Metadata{
+			Name:    fmt.Sprintf(`"%s"."test"`, keyspace),
+			Columns: []string{"id", "data"},
+			PartKey: []string{"id"},
+		}),
+		data: func() []*TestData {
+			lst := make([]*TestData, nRows)
+			for i := 0; i < nRows; i++ {
+				lst[i] = &TestData{Id: i, Data: rand.String(32)}
+			}
+			return lst
+		}(),
+		replicationFactor: replicationFactor,
+	}
+
+	var err error
+	di.session, err = di.createSession(ctx, sc)
+	if err != nil {
+		return nil, fmt.Errorf("can't create session: %w", err)
+	}
+
+	return di, nil
+}
+
+func (di *DataInserter) Close() {
+	di.session.Close()
+}
+
+// UpdateClientEndpoints closes an existing session and opens a new one.
+// In case an error was returned, DataInserter can no Longer be used.
+func (di *DataInserter) UpdateClientEndpoints(ctx context.Context, sc *scyllav1.ScyllaCluster) error {
+	di.session.Close()
+
+	session, err := di.createSession(ctx, sc)
```

Comment:
nit: I think I'd just assign directly into `di.session` so the structure reflects the new one in case there is an error and it would be accidentally accessed